src/frequenz/sdk/timeseries/formula_engine/_formula_steps.py

```diff
@@ -343,6 +347,41 @@ def apply(self, eval_stack: list[float]) -> None:
         eval_stack.append(val)
 
 
+class FallbackMetricFetcher(ABC, Generic[QuantityT]):
```

Comment:
For new classes I recommend inheriting directly from `BackgroundService`.

Reply:
Wow I didn't know we have this class, thanks :) 
Fixed

Reply:
Yeah, and it is about to be replaced by a new one already :laughing: 

* https://frequenz-floss.github.io/frequenz-core-python/latest/reference/frequenz/core/asyncio/#frequenz.core.asyncio.Service
*https://frequenz-floss.github.io/frequenz-core-python/latest/reference/frequenz/core/asyncio/#frequenz.core.asyncio.ServiceBase